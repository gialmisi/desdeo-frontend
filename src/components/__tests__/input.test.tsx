import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import InputButton from '../InputButton';


// simple tests to test rendering, property values and functions
test('should render InputButton', () => {
    render(<InputButton
        stepNumber={1}
        handleChange={() => { }} />);
    const inputele = screen.getByTestId('inputbutton');
    expect(inputele).toBeInTheDocument();
    //expect(inputele).toHaveProperty('stepNumber')
})



// test snapshots. Remember if changing stuff e.g props, snapshot will not be the same,
// hence test will fail even though should not. Update snapshot then.
test('matches snapshot', () => {
    const tree = renderer.create(<InputButton
        stepNumber={1}
        handleChange={() => { }} />).toJSON();

    expect(tree).toMatchSnapshot();
})



