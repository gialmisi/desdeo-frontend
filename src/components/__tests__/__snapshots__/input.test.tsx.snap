// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  action=""
  className=""
  data-testid="inputbutton"
  onSubmit={[Function]}
>
  <div>
    <input
      className="form-control"
      defaultValue={1}
      name="values"
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Set
  </button>
</form>
`;
